### PR TITLE
Initial implementation of PhysMem

### PR DESCRIPTION
**Pull request by Mirageinvo**
- Added PhysMem class (initial implementation).
- Added main entrypoint in simulator/simulator.cpp.
- Added some useful defines to configs/macros.hpp.

### CODE REVIEW

**Comment by Mirageinvo** [comment]
Left straightforward returning of std::vector pointer in initial commit, because it needed to check elf loader implementation to continue

File: simulator/CMakeLists.txt
Comment (MrYar0s) [inline simulator/CMakeLists.txt:1]: Please, move tests into separate directory in memory and delete main

Comment (Mirageinvo) [reply simulator/CMakeLists.txt:1]: Done

File: memory/physical_ram/phys_ram.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#include "phys_ram.hpp"
+
+namespace simulator::mem {
+PhysMem::PhysMem(uint32_t page_size, uint64_t total_size): page_size_(page_size), total_size_(total_size) {
```
[the remark this answers is not in the document]
Comment (Mirageinvo) [reply memory/physical_ram/phys_ram.cpp]: Done

File: memory/physical_ram/phys_ram.cpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    //TODO(Mirageinvo): change uint32_t's size to Page class's size
+    //TODO(Mirageinvo): check that page_size_ is a power of 2
+    pages_.resize(initial_page_number * page_size_ / sizeof(uint32_t));
+}
+
+/* static */
+PhysMem* PhysMem::CreatePhysMem(uint32_t page_size, uint64_t total_size) {
+    assert((page_size > 0) && (total_size > 0));
```
[the remark this answers is not in the document]
Comment (Mirageinvo) [reply memory/physical_ram/phys_ram.cpp]: Changed to assert(total_size != 0)

**Review by PozhiloyPumba — APPROVED** [review]

**Review by MrYar0s — APPROVED** [review]

**Pull request merged**